### PR TITLE
Create workload pack group installers

### PR DESCRIPTION
**Pull request by dsplaisted**

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks.Workloads/src/CreateVisualStudioWorkload.wix.cs [hunk 1]
```diff
@@ -215,13 +248,53 @@ public override bool Execute()
...
+                                        buildData.Remove(sourcePackage);
+                                    }
+                                }
+                            }
+
+                            string packGroupId = null;
+
+                            if (CreateWorkloadPackGroups)
+                            {
+                                //  TODO: Support passing in data to skip creating pack groups for certain packs (possibly EMSDK, because it's large)
```
Comment (joeloff) [inline src/Microsoft.DotNet.Build.Tasks.Workloads/src/CreateVisualStudioWorkload.wix.cs:271]: Does this mean EMSDK would merge its packs if they used this feature, so for now they just run with `CreateWorkloadPackGroups=false`?

Comment (joeloff) [reply src/Microsoft.DotNet.Build.Tasks.Workloads/src/CreateVisualStudioWorkload.wix.cs:271]: I wonder if we could consider a `groupable` property in the manifest files, or perhaps a `groupId` that can guide the tools. Another solution eventually would be to allow owners to pass in a set of items that contain pack IDs that can be grouped by the build task

File: src/Microsoft.DotNet.Build.Tasks.Workloads.Tests/SwixComponentTests.cs [hunk 2]
```diff
@@ -23,7 +23,7 @@ public void ItAssignsDefaultValues()
         {
```
Comment (joeloff) [inline src/Microsoft.DotNet.Build.Tasks.Workloads.Tests/SwixComponentTests.cs:23]: Can we add a test to verify the dependencies in the generated .swr when the packGroupId is passed?

File: src/Microsoft.DotNet.Build.Tasks.Workloads/src/Msi/MsiBase.wix.cs [hunk 3]
```diff
@@ -151,12 +152,12 @@ protected CompilerToolTask CreateDefaultCompiler()
 
             candle.AddPreprocessorDefinition(PreprocessorDefinitionNames.EulaRtf, GenerateEula());
             candle.AddPreprocessorDefinition(PreprocessorDefinitionNames.Manufacturer, Manufacturer);
-            candle.AddPreprocessorDefinition(PreprocessorDefinitionNames.PackageId, Package.Id);
-            candle.AddPreprocessorDefinition(PreprocessorDefinitionNames.PackageVersion, $"{Package.PackageVersion}");
+            candle.AddPreprocessorDefinition(PreprocessorDefinitionNames.PackageId, Metadata.Id);
+            candle.AddPreprocessorDefinition(PreprocessorDefinitionNames.PackageVersion, $"{Metadata.PackageVersion}");
             candle.AddPreprocessorDefinition(PreprocessorDefinitionNames.Platform, Platform);
             candle.AddPreprocessorDefinition(PreprocessorDefinitionNames.ProductCode, $"{Guid.NewGuid()}");
```
Comment (joeloff) [inline src/Microsoft.DotNet.Build.Tasks.Workloads/src/Msi/MsiBase.wix.cs:158]: Can we fix the GUID here instead? That way anything else that use the variable now and in the future will have the same value. We can just change it to be `{Guid.NewGuid():B}`

File: src/Microsoft.DotNet.Build.Tasks.Workloads/src/CreateVisualStudioWorkload.wix.cs (hunk 1 above)
Comment (dsplaisted) [reply src/Microsoft.DotNet.Build.Tasks.Workloads/src/CreateVisualStudioWorkload.wix.cs:271]: I was thinking that we would have a parameter which would be a list of packs *not* to group together. But if we only need it for Emscripten SDK, then maybe that's overkill and we should just use the `CreateWorkloadPackGroups=false` solution.

**Comment by joeloff** [comment]
We'll probably need @mmitche to help with the merge

**Pull request merged**